packages/stryker-webpack-transpiler/src/fs/OutputFileSystem.ts

```diff
@@ -0,0 +1,94 @@
+import { PathLike } from 'fs';
+import * as path from 'path';
+import { webpack, EmptyCallback, Callback } from '../types';
+import { BinaryFile, TextFile, FileKind } from 'stryker-api/core';
+import ChunkSorter, { Chunk } from '../compiler/ChunkSorter';
+
+const binaryFileExtensions = Object.freeze(['.ico', '.zip']);
+
+export default class OutputFileSystem implements webpack.OutputFileSystem {
+
+  private _files: {
+    [name: string]: string | Buffer;
+  };
+
+  constructor() {
+    this.purge();
+  }
+
+  public purge() {
+    this._files = Object.create(null);
+  }
+
+  private isBinary(fileName: string) {
+    return binaryFileExtensions.indexOf(path.extname(fileName)) >= 0;
+  }
+
+  public collectFiles(chunks: Chunk[]): Array<BinaryFile | TextFile> {
+    const files: Array<BinaryFile | TextFile> = [];
+    Object.keys(this._files).forEach(fileName => {
+      const fileContent = this._files[fileName];
+      // Best guess the file properties based on the extension
+      // Far from ideal, but i don't know any other way.
+      if (this.isBinary(fileName) && typeof fileContent !== 'string') {
+        files.push({
+          name: fileName,
+          kind: FileKind.Binary,
+          mutated: false,
+          included: false,
+          transpiled: true,
+          content: fileContent
+        });
+      } else {
+        files.push({
+          name: fileName,
+          kind: FileKind.Text,
+          mutated: true,
+          included: true,
+          transpiled: true,
+          content: fileContent.toString()
+        });
+      }
+    });
+
+    const sortedFiles = new ChunkSorter()
+      .sortedFileNames(chunks);
+    files.sort((a, b) => {
```

Comment:
> Is this an alphanumeric sort

No, this is the order based on the index of the `sortedFiles` array (which is the topological order from the `ChunkSorter`).

```ts
files.sort((a, b) => {
      const aName = path.basename(a.name);
      const bName = path.basename(b.name);
      return sortedFiles.indexOf(aName) - sortedFiles.indexOf(bName);
    });
```

> But you could configure bundels to end up in different folders, right?

Yes, it could. And the ChunkSorter will be able to handle that.